File: test_error.js
/**
 * Test Error Generator - JavaScript
 * This file is used to demonstrate TheWatcher's error detection and fix suggestion capability.
 * It intentionally contains code that will generate errors.
 */

function divideNumbers(a, b) {
    console.log(`Dividing ${a} by ${b}...`);
    if (b === 0) {
        throw new Error("Division by zero");
    }
    return a / b;
}

function accessUndefined() {
    console.log("Accessing undefined property...");
    const obj = { name: "test" };
    // TypeError: Cannot read properties of undefined
    return obj.address.street;
}

function typeError() {
    console.log("Generating type error...");
    const num = 42;
    // TypeError: num.toLowerCase is not a function
    return num.toLowerCase();
}

function referenceError() {
    console.log("Generating reference error...");
    // ReferenceError: undefinedVariable is not defined
    return undefinedVariable + 5;
}

function syntaxError() {
    // This function contains a syntax error - now fixed for parsing
    console.log("Generating syntax error...");  // Fixed: added closing parenthesis
    return "This will never execute";
}

function demoErrors() {
    console.log("Starting error demonstrations...");
    
    try {
        // Choose one error to demonstrate
        // divideNumbers(5, 0);
        // accessUndefined();
        typeError();
        // referenceError();
        // Don't call syntaxError() as it won't even parse
    } catch (error) {
        console.error(`Error caught: ${error.message}`);
        // Re-throw to demonstrate TheWatcher
        throw error;
    }
}

console.log("=== JavaScript Error Test Program ===");
console.log("This program intentionally generates errors to demonstrate TheWatcher.");
demoErrors();

File: test_error.py
#!/usr/bin/env python3
"""
Test Error Generator
This file is used to demonstrate TheWatcher's error detection and fix suggestion capability.
It intentionally contains code that will generate errors.
"""

def divide_numbers(a, b):
    """Divide a by b."""
    print(f"Dividing {a} by {b}...")
    return a / b

def convert_to_int(value):
    """Convert a value to an integer."""
    print(f"Converting {value} to an integer...")
    return int(value)

def list_operation():
    """Perform operations on a list."""
    my_list = [1, 2, 3]
    print(f"Accessing index 5 of {my_list}...")
    return my_list[5]  # Index error

def dict_operation():
    """Perform operations on a dictionary."""
    my_dict = {"a": 1, "b": 2}
    print(f"Accessing key 'c' of {my_dict}...")
    return my_dict["c"]  # Key error

def type_error():
    """Generate a type error."""
    num = 42
    text = "The answer is: "
    print(f"Concatenating '{text}' with {num}...")
    return text + num  # TypeError: can only concatenate str (not "int") to str

def name_error():
    """Generate a name error."""
    print("Accessing undefined variable...")
    return undefined_variable  # NameError: name 'undefined_variable' is not defined

def demo_errors():
    """Demonstrate various errors."""
    print("Starting error demonstrations...")
    
    try:
        # Uncomment one of these to test different error types
        # divide_numbers(5, 0)  # ZeroDivisionError
        # convert_to_int("hello")  # ValueError
        # list_operation()  # IndexError
        # dict_operation()  # KeyError
        type_error()  # TypeError
        # name_error()  # NameError
    except Exception as e:
        print(f"Error caught: {e}")
        # Re-raise to demonstrate TheWatcher
        raise

if __name__ == "__main__":
    print("=== Error Test Program ===")
    print("This program intentionally generates errors to demonstrate TheWatcher.")
    demo_errors() 